pkg/trie/triedb/child_tries.go

```diff
@@ -8,14 +8,14 @@ import (
 	"github.com/ChainSafe/gossamer/pkg/trie"
 )
 
-func (t *TrieDB) GetChild(keyToChild []byte) (trie.Trie, error) {
+func (t *TrieDB[H, Hasher]) GetChild(keyToChild []byte) (trie.Trie, error) {
 	panic("not implemented yet")
```

Comment:
Based on how substrate trie works and our previous discussions I think we aren't going to implement this methods as part of `TrieDB`. We are going to use a new layer on top of TrieDB to handle the child tries.
Perhaps we can create a ticket to track this and remove this methods from the Trie interface when we start using the TrieDB

Reply:
👍 I'll remove this file completely.  We already have an issue to track child trie support in future work.  #3836